clusterloader2/pkg/measurement/common/slos/windows_node_resource_usage.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package slos
+
+import (
+	"math"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	windowsResourceUsagePrometheusMeasurementName = "WindowsResourceUsagePrometheus"
+	// get top 10 non-system processes with highest cpu usage within 1min query window size
+	cpuUsageQueryTop10 = `topk(10, sum by (process) (rate(wmi_process_cpu_time_total{process!~"Idle|Total|System"}[1m]) / on(job) group_left wmi_cs_logical_processors) * 100)`
```

Comment:
IIUC, this query has unbounded timeframe meaning that it will be always taking into account all metrics since the prometheus server was started. Usually we try to limit the window to the actual timeframe of the test. If you run your test multiple times (either run CL2 multiple times without tearing down prometheus server or run a suite of multiple windows tests  in single CL2 invocation) this measurements will not work as expected. Could you fix it? I'm not familiar with the promQL syntax you used here (is it something new?), but maybe you can get some inspiration comparing how it's done in api-call-latency measurement - https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus.go

Reply:
1. I see, and CMIIW, I think I set it as 1min static bounded timeframe, as I wanted to collect the peak usage when all pods are running.
2. As I need to divide by logical processors which is a constant. If divided directly there was syntax error as it didn't work with rate() function, so I spent some time and found that using the many-to-one left join works fine. Reference: https://prometheus.io/docs/prometheus/latest/querying/operators/#many-to-one-and-one-to-many-vector-matches